### ISSUE TITLE
test failure: "should handle unknown arguments correctly" fails on non-english locale

### ISSUE THREAD

**Issue by mkg20001** [ipfs/js-ipfs#1792]
- **Version**: master 7bcc49615e54f09c34bae07c994e3be3d870fe1f
- **Platform**: ubuntu 18.10 amd64, node v10.14.2
- **Subsystem**: cli

#### Steps to reproduce the error:

Run tests with the following environment variables set 
```
LANG=de_DE.UTF-8
LANGUAGE=de_DE.UTF-8
LC_CTYPE="de_DE.UTF-8"
LC_NUMERIC="de_DE.UTF-8"
LC_TIME="de_DE.UTF-8"
LC_COLLATE="de_DE.UTF-8"
LC_MONETARY="de_DE.UTF-8"
LC_MESSAGES="de_DE.UTF-8"
LC_PAPER="de_DE.UTF-8"
LC_NAME="de_DE.UTF-8"
LC_ADDRESS="de_DE.UTF-8"
LC_TELEPHONE="de_DE.UTF-8"
LC_MEASUREMENT="de_DE.UTF-8"
LC_IDENTIFICATION="de_DE.UTF-8"
LC_ALL="de_DE.UTF-8"
```

Will produce following crash

```
  1203 passing (32m)
  117 pending
  1 failing

  1) cli
       general cli options
         daemon off (directly to core)
           should handle unknown arguments correctly:
     AssertionError: expected 'Unbekannte Argumente: again, random\nbin.js <command>\n\nKommandos:\n  bin.js files <command>          Operations over mfs files (ls, mkdir, rm, etc)\n  bin.js add <file...>            Add a file to IPFS using the UnixFS data\n                                  format\n  bin.js bitswap <command>        Interact with the bitswap agent.\n  bin.js block <command>          Manipulate raw IPFS blocks.\n  bin.js bootstrap <command>      Show or edit the list of bootstrap peers.\n  bin.js cat <ipfsPath>           Fetch and cat an IPFS path referencing a file\n  bin.js cid <command>            Convert, format and discover properties of\n                                  CIDs.\n  bin.js commands                 List all available commands\n  bin.js config <key> [value]     Get and set IPFS config values\n  bin.js daemon                   Start a long-running daemon process\n  bin.js dag <command>            Interact with ipld dag objects.\n  bin.js dns <domain>             Resolve DNS links\n  bin.js file <command>           Interact with IPFS objects representing Unix\n                                  filesystems.\n  bin.js get <ipfsPath>           Fetch a file or directory with files\n                                  references from an IPFS Path\n  bin.js id                       Shows IPFS Node ID info\n  bin.js init [config] [options]  Initialize a local IPFS node\n  bin.js key                      Manage your keys\n  bin.js ls <key>                 List files for the given directory\n  bin.js name <command>           Publish and resolve IPNS names.\n  bin.js object <command>         Interact with ipfs objects.\n  bin.js pin                      Pin and unpin objects to local storage.\n  bin.js ping <peerId>            Measure the latency of a connection\n  bin.js pubsub <command>         pubsub commands\n  bin.js repo <command>           Manipulate the IPFS repo.\n  bin.js resolve <name>           Resolve the value of names to IPFS\n  bin.js shutdown                 Shut down the ipfs daemon\n  bin.js stats <command>          Query IPFS statistics.\n  bin.js swarm <command>          Swarm inspection tool.\n  bin.js version                  Shows IPFS version information\n  bin.js completion               generate bash completion script\n\nOptionen:\n  --version  Version anzeigen                                          [boolean]\n  --silent   Write no output                         [boolean] [Standard: false]\n  --pass     Pass phrase for the keys                    [string] [Standard: ""]\n  --help     Hilfe anzeigen                                            [boolean]\n\nipfs uses a repository in the local file system. By default, the repo is located\nat ~/.jsipfs. To change the repo location, set the $IPFS_PATH environment\nvariable:\n\nexport IPFS_PATH=/path/to/ipfsrepo\n' to include 'Unknown arguments: again, random'
      at thing.ipfs.then (test/cli/general.js:16:22)
      at process._tickCallback (internal/process/next_tick.js:68:7)



Command failed: mocha --timeout 10000 --ui bdd --colors --exit test/node.js test/**/*.spec.js
null
null
Error: Command failed: mocha --timeout 10000 --ui bdd --colors --exit test/node.js test/**/*.spec.js
null
null
    at makeError (/home/maciej/Projekte/js-ipfs/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/home/maciej/Projekte/js-ipfs/node_modules/execa/index.js:278:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
npm ERR! Test failed.  See above for more details.
```

Reason: Yargs seems to be translated in that language, automatically uses translated strings

**Comment by vincepmartin** [comment]
I started looking into this and had some questions about the behavior that we want from this test and imagine we could tackle this in two different ways.

1. We could attempt to make Yargs operate in the EN locale during testing, regardless of the true set locale.  This would allow for us to keep the hard coded String that is expected in the test.

or

2. We could try to get the EN string in the test to be translated to the proper language dynamically. This would be more complex than the first option.

Thoughts?

**Comment by mkg20001** [comment]
I'd go for 1. and then add a test to specifically test localization with only a small subset of strings, once implemented. This would keep overall overhead lowest, especially since some languages contain utf8 characters which might cause unexpected troubles with some length comparisons.

**Comment by mkg20001** [comment]
Btw, this is how you really make sure it detects english ;)
```js
Object.assign(process.env, {"LANG": "en_US.UTF-8",
"LANGUAGE": "en_US.UTF-8",
"LC_CTYPE": "en_US.UTF-8",
"LC_NUMERIC": "en_US.UTF-8",
"LC_TIME": "en_US.UTF-8",
"LC_COLLATE": "en_US.UTF-8",
"LC_MONETARY": "en_US.UTF-8",
"LC_MESSAGES": "en_US.UTF-8",
"LC_PAPER": "en_US.UTF-8",
"LC_NAME": "en_US.UTF-8",
"LC_ADDRESS": "en_US.UTF-8",
"LC_TELEPHONE": "en_US.UTF-8",
"LC_MEASUREMENT": "en_US.UTF-8",
"LC_IDENTIFICATION": "en_US.UTF-8",
"LC_ALL": "en_US.UTF-8"})
```

_(`LC_ALL` usually is enough, **but you can never be sure**)_

**Comment by vincepmartin** [comment]
Thanks!  I was just setting it in my .profile at the moment :)

**Comment by mkg20001** [comment]
@vincepmartin You can simply prepend the environment variables to the command. For ex `LC_ALL=de_DE.UTF-8 aegir test` or if that doesn't work then `env LC_ALL=de_DE.UTF-8 aegir test`. No need for touching `.profile`

**Comment by vincepmartin** [comment]
I think I have this one worked out.  Not sure if it is an acceptable fix or not.  It now passes all tests regardless of locale :)

**Comment by achingbrain** [comment]
This should be fixed now and will be released in `v0.39.0`, please reopen if it still occurs.

**Comment by vincepmartin** [comment]
Yippie!  My first contrib ever to an open source project :)
